src/hypertable_restrict_info.c

```diff
@@ -524,6 +549,18 @@ gather_restriction_dimension_vectors(const HypertableRestrictInfo *hri)
 														   open->lower_strategy,
 														   open->lower_bound);
 
+				/*
+				 * If we have a nontrivial condition for the second index column
+				 * range_start, use a backward scan direction, so that the index
+				 * is able to use the second column as well to choose the
+				 * starting point.
+				 * If not, prefer forward direction, because backwards scan is
+				 * slightly slower for some reason.
+				 * Ideally we need some other index type than btree for this.
```

Comment:
Why do you need non-btree index here, what would we gain by other index type.

Reply:
The btree index is not so suitable for queries like `find interval that contains the given point` and `find intervals that intersect with the given interval`. Idk what it should be, we also have this dimension id column on which we should filter first.

Reply:
I mean, in all queries we make the third column (range_end) is only compared in a sequential scan over index entries, not used to find the starting/ending point.